File: projects/03/letter_grade.rb
#! /usr/bin/ruby
# letter_grade.rb translates a score into a grade
# Completed by Justin Voss, 1/28/22
################################################################

# Input:  A score
# Precondition: The score is an integer
# Output: The corresponding letter grade

#  Replace this line with the definition of function yearCode() 
def letterGrade(score)
   case (score/10)
   when 10
        "A"
   when 9
        "A"
   when 8
        "B"
   when 7
        "C"
   when 6
        "D"
   else 
        "F"
 end 
end 

 
 if __FILE__ == $0
    print "Enter the score: "
    score = gets.chomp.to_i  #turns the input into an integer
    print "Grade is: "
    puts letterGrade(score)
 end

File: projects/03/clojure/src/letter_grade.clj
;;;; letter_grade.clj is a driver for function letterGrade.
;;;;
;;;; Input: A score, an integer
;;;; Output: The grade, either A, B, C, D, or F.
;;;;
;;;; Usage: clojure -m year_codes
;;;;

;;;; Completed by: Justin Voss
;;;; Date: 1/28/22
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grade)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; letterGrade() returns the grade.
;;; Receive: score, an integer
;;; Precondition: it is an integer
;;; Return: the corresponding letter grade
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;





;; solution using the cond function
(defn letterGrade [score]
(let [score ( / score 10)]
  (cond
    (= score 10) "A"
    (= score 9)  "A"
    (= score 8)  "B"
    (= score 7)  "C"
    (= score 6)  "D"
    :else                "F"
  ))
)
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our letterGrade() function.
;;; Input: score, an integer
;;; Output: the corresponding letter grade
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter score: ") (flush)
 (let
    [ score (int(read)) ]                ; read score (an integer)
    (println (letterGrade score) "\n")      ; display its grade
 )
)